test/linalg.jl

```diff
@@ -245,6 +246,23 @@ end
             end
         end
     end  # SVD
+    @testset "Left multiplication" begin
+        N1 = 20
+        N2 = 8
+        N3 = 5
+        N4 = N1 + N3 - N2
+        A = BlockDiagonal([rand(rng, N1, N1), rand(rng, N2, N2)])
+        B = BlockDiagonal([rand(rng, N1, N2), rand(rng, N3, N4)])
+        x = rand(rng, N1 + N2)
+        y = rand(rng, N2 + N4)
+
+        @testset "Lower triangular" begin
+            @test lmul!(LowerTriangular(A), copy(x)) ≈ lmul!(LowerTriangular(Matrix(A)), copy(x))
+            @test lmul!(LowerTriangular(B), copy(y)) ≈ lmul!(LowerTriangular(Matrix(B)), copy(y))
+            cx = copy(x)
+            @test 2*192 > @ballocated lmul!($(LowerTriangular(A)), $cx) # give +100% leeway
```

Comment:
why the leeway?

Reply:
Just so it doesn't break if allocations change slightly. Could change to?
```suggestion
            @test 192 => @ballocated lmul!($(LowerTriangular(A)), $cx)
```